crates/wallet/src/wallet/tx_builder.rs

```diff
@@ -42,6 +42,9 @@ use alloc::{boxed::Box, rc::Rc, string::String, vec::Vec};
 use core::cell::RefCell;
 use core::fmt;
 
+use std::sync::Arc;
```

Comment:
🔧 Need to change the `Arc` type to be compatible with no-std (and WASM).

```suggestion
use alloc::sync::Arc;
```